### PR TITLE
SNOW-1625536: Add more options to save_as_table

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1625536

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Review by sfc-gh-jrose — APPROVED** [review]

File: src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py [hunk 1]
```diff
@@ -752,6 +757,14 @@ def get_comment_sql(comment: Optional[str]) -> str:
     )
 
 
+def get_assign_param_sql(param_name: str, param_value: Optional[Any]) -> str:
+    return (
+        f"{param_name}{EQUALS}{param_value}"
+        if param_value is not None
+        else EMPTY_STRING
+    )
```
Comment (sfc-gh-jrose) [inline src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py:765]: Could `get_options_statement` be used in place of adding a new function?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,12 @@
 
 #### New Features
 - Added support for `snowflake.snowpark.testing.assert_dataframe_equal` that is a util function to check the equality of two Snowpark DataFrames.
+- Added support for specifying the following attributes when creating table using `DataFrameWriter.save_as_table`:
+  - `enable_schema_evolution`
+  - `data_retention_time`
+  - `max_data_extension_time`
+  - `chagne_tracking`
```
Comment (sfc-gh-yixie) [inline CHANGELOG.md:13]: change_tracking. A typo.

File: src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py [hunk 3]
```diff
@@ -771,11 +789,28 @@ def create_table_statement(
...
+        + LEFT_PARENTHESIS
+        + schema
+        + RIGHT_PARENTHESIS
+        + cluster_by_clause
+        + get_assign_param_sql(ENABLE_SCHEMA_EVOLUTION, enable_schema_evolution)
+        + get_assign_param_sql(DATA_RETENTION_TIME_IN_DAYS, data_retention_time)
+        + get_assign_param_sql(MAX_DATA_EXTENSION_TIME_IN_DAYS, max_data_extension_time)
+        + get_assign_param_sql(CHANGE_TRACKING, change_tracking)
+        + (COPY_GRANTS if copy_grants else EMPTY_STRING)
+        + comment_sql
```
Comment (sfc-gh-yixie) [inline src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py:813]: Suggest formatting string

File: src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py (hunk 1 above)
Comment (sfc-gh-aalam) [reply src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py:765]: that's for pointing this out. I didn't know this existed

**Review by sfc-gh-yixie — APPROVED** [review]

**Pull request merged**